docker/ci/determine_country.bash

```diff
@@ -0,0 +1,130 @@
+if ! which netselect >/dev/null; then
+	wget --progress=bar -O netselect.deb 'http://ftp.debian.org/debian/pool/main/n/netselect/netselect_0.3.ds1-28+b1_amd64.deb'
+	sudo dpkg -i netselect.deb
+	rm netselect.deb
+fi
+
+cn_mirrors=(
+	mirrors.aliyun.com
+	mirrors.ustc.edu.cn
+	mirrors.163.com
+)
+en_mirrors=(
+	archive.ubuntu.com
+	ftp.de.debian.org
+	packages.debian.org
+)
+
+function get_speed_score() {
+	sudo netselect ${@} | awk '{print $1}'
+}
+
+echo "Guessing if we are in China..."
+if [[ $(get_speed_score ${cn_mirrors[@]}) -lt $(get_speed_score ${en_mirrors[@]} ) ]]; then
+	echo "Assume we are in China."
+	export WE_ARE_IN_CHINA=true
+fi
+
+if [[ ! "${WE_ARE_IN_CHINA}" ]]; then
+	exit 0
+fi
+
+function set_apt_mirror() {
+	sudo cat <<EOF >/etc/apt/sources.list
+deb http://mirrors.aliyun.com/ubuntu/ bionic main restricted universe multiverse
+deb http://mirrors.aliyun.com/ubuntu/ bionic-security main restricted universe multiverse
+deb http://mirrors.aliyun.com/ubuntu/ bionic-updates main restricted universe multiverse
+deb http://mirrors.aliyun.com/ubuntu/ bionic-proposed main restricted universe multiverse
+deb http://mirrors.aliyun.com/ubuntu/ bionic-backports main restricted universe multiverse
+deb-src http://mirrors.aliyun.com/ubuntu/ bionic main restricted universe multiverse
+deb-src http://mirrors.aliyun.com/ubuntu/ bionic-security main restricted universe multiverse
+deb-src http://mirrors.aliyun.com/ubuntu/ bionic-updates main restricted universe multiverse
+deb-src http://mirrors.aliyun.com/ubuntu/ bionic-proposed main restricted universe multiverse
+deb-src http://mirrors.aliyun.com/ubuntu/ bionic-backports main restricted universe multiverse
+EOF
+	sudo apt-get -qq update
+}
+
+function set_go_proxy() {
+	# other option: export GOPROXY=https://mirrors.aliyun.com/goproxy/
+	export GOPROXY=https://goproxy.cn
+}
+
+function set_maven_repo() {
+	mirror="<mirrors>
+      <mirror>
+        <id>alimaven</id>
+        <mirrorOf>central</mirrorOf>
+        <name>aliyun maven</name>
+        <url>http://maven.aliyun.com/nexus/content/repositories/central/</url>
+      </mirror>
+	</mirrors>"
+	mkdir -p "$HOME/.m2"
+	settings="$HOME/.m2/settings.xml"
+	if [[ ! -f "${settings}" ]]; then
+		cat <<EOF > "${settings}"
+<?xml version="1.0" encoding="UTF-8"?>
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
+	${mirror}
+</settings>
+EOF
+	else
+		mirror=`echo "${mirror}" | sed -e 's/"/\\"/g' -e 's^\\/^\\\\/^g'`
+		mirror=${mirror//$'\n'/'\n'}
+		if grep -q -o "<mirrors>" "${settings}"; then
+			sed -i "/<mirrors>/,/<\/mirrors>/c${mirror}" "${settings}"
+		else
+			sed -i "s^</settings>^${mirror}\n&^" "${settings}"
+		fi
+	fi
+}
+
+args=`getopt -o "" -l"set-apt-mirror,set-go-proxy,set-maven-repo" -- "$@"`
+eval set -- "${args}"
+while true; do
```

Comment:
Let us modulize the code by having 

- find_fastest_maven_repo
- find_fastest_apt_source
- find_fastest_go_proxy

Other than have one script that does three things.

Reply:
OK, I will do that